layers.go

```diff
@@ -776,8 +777,17 @@ func (r *layerStore) Mount(id string, options drivers.MountOpts) (string, error)
 		return "", ErrLayerUnknown
 	}
 	if layer.MountCount > 0 {
-		layer.MountCount++
-		return layer.MountPoint, r.saveMounts()
+		mounted, err := mount.Mounted(layer.MountPoint)
+		if err != nil {
+			return "", err
+		}
+		// If the container is not mounted then we have a condition
+		// where the kernel umounted the mount point. This means
+		// that the mount count never got decremented.
+		if mounted {
+			layer.MountCount++
```

Comment:
shouldn't we still increase the counter when we mount it again?

Reply:
I don't think so, since we know that it was unmounted once, that we did not take into account.  At this point the mount count is useless, since we don't know if their are any processes still out their that have mounted the image.  In this case we are likely to leak the mount or at least force the user to have to unmount it more then is necessary to get the mount point to unmount.

The mount count was always an estimation of how many processes needed to mount the storage.

Reply:
And the mountcount is incremented below. when it falls through to mount the layer